src/internal/testing/ColdObservable.ts

```diff
@@ -1,35 +1,30 @@
 import { Observable } from '../Observable';
 import { Subscription } from '../Subscription';
-import { Scheduler } from '../Scheduler';
 import { TestMessage } from './TestMessage';
-import { SubscriptionLog } from './SubscriptionLog';
-import { SubscriptionLoggable } from './SubscriptionLoggable';
-import { applyMixins } from '../util/applyMixins';
 import { Subscriber } from '../Subscriber';
 import { observeNotification } from '../Notification';
+import { SchedulerLike, TeardownLogic } from '../types';
+import { logSubscribedFrame, logUnsubscribedFrame, SubscriptionLog } from './subscription-logging';
 
-export class ColdObservable<T> extends Observable<T> implements SubscriptionLoggable {
+export class ColdObservable<T> extends Observable<T> {
   public subscriptions: SubscriptionLog[] = [];
-  scheduler: Scheduler;
-  // @ts-ignore: Property has no initializer and is not definitely assigned
-  logSubscribedFrame: () => number;
-  // @ts-ignore: Property has no initializer and is not definitely assigned
-  logUnsubscribedFrame: (index: number) => void;
+  logSubscribedFrame = logSubscribedFrame;
+  logUnsubscribedFrame = logUnsubscribedFrame;
 
-  constructor(public messages: TestMessage[], scheduler: Scheduler) {
-    super(function (this: Observable<T>, subscriber: Subscriber<any>) {
```

Comment:
Moved this super argument out to just be declared as a method. It's just less complicated.